pkg/resource/bucket/custom_acl_code.go

```diff
@@ -0,0 +1,216 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package bucket
+
+import (
+	"fmt"
+	"strings"
+
+	svcsdk "github.com/aws/aws-sdk-go/service/s3"
+)
+
+// Only some of these exist in the SDK, so duplicating them all here
+var (
+	CANNED_ACL_PRIVATE               = "private"
+	CANNED_PUBLIC_READ               = "public-read"
+	CANNED_PUBLIC_READ_WRITE         = "public-read-write"
+	CANNED_AWS_EXEC_READ             = "aws-exec-read"
+	CANNED_AUTHENTICATED_READ        = "authenticated-read"
+	CANNED_BUCKET_OWNER_READ         = "bucket-owner-read"
+	CANNED_BUCKET_OWNER_FULL_CONTROL = "bucket-owner-full-control"
+	CANNED_LOG_DELIVERY_WRITE        = "log-delivery-write"
+)
+
+var (
+	GRANTEE_ZA_TEAM_ID              = "6aa5a366c34c1cbe25dc49211496e913e0351eb0e8c37aa3477e40942ec6b97c"
+	GRANTEE_LOG_DELIVERY_URI        = "http://acs.amazonaws.com/groups/s3/LogDelivery"
+	GRANTEE_ALL_USERS_URI           = "http://acs.amazonaws.com/groups/global/AllUsers"
+	GRANTEE_AUTHENTICATED_USERS_URI = "http://acs.amazonaws.com/groups/global/AuthenticatedUsers"
+)
+
+var (
+	HEADER_USER_ID_FORMAT = "id=%s"
+	HEADER_URI_FORMAT     = "uri=%s"
+)
```

Comment:
I actually prefer the `ALL_CAPS` style for constants, but the Go style convention is actually to use `CamelCase` for exported variable names (constant or otherwise) and `lowerCamelCase` for unexported variable names. It's just a nit, but consider (even in a future commit if you want) unexporting all of these constants and following the `lowerCamelCase` naming convention. Same goes for the previous `ALL_CAPS`-style constants in the hooks.go file...

Reply:
Oh lame. All caps is far superior. Will move these all into lowerCamelCase ;(